### PR TITLE
Add Mellow Fly E3 V2

### PR DESCRIPTION
**Pull request by jaysuk**
### Description

Add the Fly-E3-v2 mainboard to Marlin

### Requirements

A Fly-E3-V2

### Benefits

Adds support for the Fly-E3-V2 board

### Related Issues

None

### CODE REVIEW

File: Marlin/src/pins/stm32f4/pins_FLY_E3_V2.h [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+ *
+ */
+#pragma once
+
+#if NOT_TARGET(STM32F4)
+  #error "Oops! Select an STM32F4 board in 'Tools > Board.'"
+#endif
+
+#ifndef BOARD_INFO_NAME
+  #define BOARD_INFO_NAME "FLY_E3_V2"
```
Comment (thisiskeithb) [inline Marlin/src/pins/stm32f4/pins_FLY_E3_V2.h]: "Mellow FLY E3 V2" would probably look a bit nicer in the firmware info screen.

File: Marlin/src/pins/stm32f4/pins_FLY_E3_V2.h [hunk 2]
```diff
@@ -0,0 +1,496 @@
...
+  #define FLASH_EEPROM_EMULATION                  // Use Flash-based EEPROM emulation
+#endif
+
+#if ENABLED(FLASH_EEPROM_EMULATION)
+  // Decrease delays and flash wear by spreading writes across the
+  // 128 kB sector allocated for EEPROM emulation.
+  #define FLASH_EEPROM_LEVELING
+#endif
+
+#define HAS_OTG_USB_HOST_SUPPORT                  // USB Flash Drive support
```
Comment (thisiskeithb) [inline Marlin/src/pins/stm32f4/pins_FLY_E3_V2.h]: Does this board really support flash drives? From the pictures I can find online, there's a USB-C port for serial comms, but no USB-A port.

File: Marlin/src/pins/stm32f4/pins_FLY_E3_V2.h (hunk 1 above)
Comment (jaysuk) [reply Marlin/src/pins/stm32f4/pins_FLY_E3_V2.h]: Happy to change

File: Marlin/src/pins/stm32f4/pins_FLY_E3_V2.h (hunk 2 above)
Comment (jaysuk) [reply Marlin/src/pins/stm32f4/pins_FLY_E3_V2.h]: and no, it doesn't. i'll remove

Comment (thisiskeithb) [reply Marlin/src/pins/stm32f4/pins_FLY_E3_V2.h]: Since this doesn't have flash drive support, those build flags should be dropped as well.

**Comment by thisiskeithb** [comment]
Looks like we have a PR for the V3 board as well, so it may make sense to combine some pins into a `*_common.h` file if there's overlap:

- https://github.com/MarlinFirmware/Marlin/pull/25894

**Comment by jaysuk** [comment]
There's no overlap

File: ini/stm32f4.ini [hunk 3]
```diff
@@ -806,3 +806,20 @@ board_build.offset = 0x8000
...
+# Mellow Fly E3 V2 (STM32F407VGT6 ARM Cortex-M4)
+#
+[env:FLY_E3_V2]
+extends                     = stm32_variant
+platform_packages           = ${stm_flash_drive.platform_packages}
+board                       = marlin_STM32F407VGT6_CCM
+board_build.variant         = MARLIN_F4x7Vx
+board_build.offset          = 0x8000
+board_upload.offset_address = 0x08008000
+build_flags                 = ${stm_flash_drive.build_flags}
```
Comment (thisiskeithb) [inline ini/stm32f4.ini]: Since this board doesn't support flash drives, this build flag should be removed.

**Comment by thinkyhead** [comment]
Is that right that this board supports the `BTT_MOTOR_EXPANSION` module?

I just received a donated Fly Mellow Super 5 Pro. I don't know how similar these Fly Mellow boards are, but it should be good for general testing. It has the STM32H723-VGT6 MCU.

**Comment by thisiskeithb** [comment]
> Is that right that this board supports the BTT_MOTOR_EXPANSION module?

These addon boards are universal and can work on any motherboard if pins are defined correctly.

Having said that, this PR was based heavily on the SKR V2 pins & environment.

**Comment by thinkyhead** [comment]
We might pass on this. https://github.com/Mellow-3D/Fly-E3-Pro-v2

"This control board has been discontinued by Flymaker/Mellow due to a number of issues that have been identified. It is recommended to stop using this printer board and return it to seller where it was purchased from for a refund."

**Comment by jaysuk** [comment]
This isn't the fly-e3-pro-v2, it's the fly-e3-v2 which is totally different.
I'm still working on this but I do have it up and running on my voron v0.2,
just away on holiday at the moment.
Last thing I'm poking at is getting USB hosting to work. I'll have it
finished some time next week.

**Comment by jaysuk** [comment]
FYI, it's this board https://teamgloomy.github.io/fly_e3_v2_general.html
I also maintain the teamgloomy wiki

**Comment by jaysuk** [comment]
I've tested this build on a voron v0.2, suing 5160's on A and B and 2209s on Z and E.
All working fine. Haven't been able to get accessing the SD card when the board is connected to a PC. M21/M22 work fine though.
Once you're happy i'll add the rest of the fly boards

**Pull request merged**